Examples/Tests/collision/analysis_collision_1d.py

```diff
@@ -0,0 +1,154 @@
+sorted_pz = data['ions', 'particle_momentum_z'][sorted_indices].value
+
+Np = len(sorted_wp)
+wpmin = sorted_wp.min();
+wpmax = sorted_wp.max();
+for i in range(len(sorted_wp)):
+    if sorted_wp[i] > wpmin:
+        Npmin = i
+        break
+
+test_tag = 'T1b'
+print(f"test_tag = {test_tag}")
+
+### use below for tests T1b and T1c, where NpA = Npmin
+if test_tag == 'T1b' or test_tag == 'T1c':
+    NpA = Npmin
+    wpA = wpmin;
+    NpB = Np - Npmin
+    wpB = wpmax;
+    NpAs = 0
+    NpAe = Npmin
+    NpBs = Npmin
+    NpBe = Np
+
+### use below for tests T1a and T1d, where NpB = Npmin
+if test_tag == 'T1a' or test_tag == 'T1d':
+    NpB = Npmin
+    wpB = wpmin;
+    NpA = Np - Npmin
+    wpA = wpmax;
+    NpBs = 0
+    NpBe = Npmin
+    NpAs = Npmin
+    NpAe = Np
+
+#############
+
+#print(f"Np = {Np}, NpA = {NpA}, NpB = {NpB}")
+#print(f"wpA = {wpA}, wpB = {wpB}")
+
+sorted_px_sum = np.abs(sorted_px).sum();
+sorted_py_sum = np.abs(sorted_py).sum();
+sorted_pz_sum = np.abs(sorted_pz).sum();
+sorted_wp_sum = np.abs(sorted_wp).sum();
+
+#print(f"sum(sorted_px)  =  {sorted_px_sum}")
+#print(f"sum(sorted_py)  =  {sorted_py_sum}")
+#print(f"sum(sorted_pz)  =  {sorted_pz_sum}")
+#print(f"sum(sorted_wp)  =  {sorted_wp_sum}")
+
+# compute mean velocities
+wAtot = wpA*NpA
+wBtot = wpB*NpB
+
+uBx = uBy = uBz = 0.
+for i in range(NpBs,NpBe):
+    uBx += wpB*sorted_px[i]
+    uBy += wpB*sorted_py[i]
+    uBz += wpB*sorted_pz[i]
+uBx /= (mass*wBtot) # [m/s]
+uBy /= (mass*wBtot) # [m/s]
+uBz /= (mass*wBtot) # [m/s]
+
+uAx = uAy = uAz = 0.
+for i in range(NpAs,NpAe):
+    uAx += wpA*sorted_px[i]
+    uAy += wpA*sorted_py[i]
+    uAz += wpA*sorted_pz[i]
+uAx /= (mass*wAtot) # [m/s]
+uAy /= (mass*wAtot) # [m/s]
+uAz /= (mass*wAtot) # [m/s]
+
+# compute temperatures
+TBx = TBy = TBz = 0.
+for i in range(NpBs,NpBe):
+    TBx += wpB*(sorted_px[i]/mass - uBx)**2
+    TBy += wpB*(sorted_py[i]/mass - uBy)**2
+    TBz += wpB*(sorted_pz[i]/mass - uBz)**2
+TBx *= mass/(e*wBtot)
+TBy *= mass/(e*wBtot)
+TBz *= mass/(e*wBtot)
+
+TAx = TAy = TAz = 0.
+for i in range(NpAs,NpAe):
+    TAx += wpA*(sorted_px[i]/mass - uAx)**2
+    TAy += wpA*(sorted_py[i]/mass - uAy)**2
+    TAz += wpA*(sorted_pz[i]/mass - uAz)**2
+TAx *= mass/(e*wAtot)
+TAy *= mass/(e*wAtot)
+TAz *= mass/(e*wAtot)
+
+TApar = TAz
+TAperp = (TAx + TAy)/2.0
+TA = (TAx + TAy + TAz)/3.0
+
+TBpar = TBz
+TBperp = (TBx + TBy)/2.0
+TB = (TBx + TBy + TBz)/3.0
+
+#print(f"uAx = {uAx/1e3:.{5}f}; uAy = {uAy/1e3:.{5}f}; uAz = {uAz/1e3:.{5}f} [km/s]")
+#print(f"uBx = {uBx/1e3:.{5}f}; uBy = {uBy/1e3:.{5}f}; uBz = {uBz/1e3:.{5}f} [km/s]")
+#print(f"TA = {TA:.{5}f}; TAperp = {TAperp:.{5}f}; TApar = {TApar:.{5}f} [eV]")
+#print(f"TB = {TB:.{5}f}; TBperp = {TBperp:.{5}f}; TBpar = {TBpar:.{5}f} [eV]")
+
+TApar_30ps_soln = 6.15e3 # TA parallel solution at t = 30 ps
+error = np.abs(TApar-TApar_30ps_soln)/TApar_30ps_soln
+tolerance = 0.02
+print('TApar at 30ps error = ', error);
+print('tolerance = ', tolerance);
```

Comment:
Could you confirm that this fails when using the WarpX version from `development` branch (i.e. without this PR) ?
Also: would it make sense to also add an `assert` checking the other quantities (`TAx`, `TAy`, as well as the `TB`s, `uA`s and `uB`s)?
And would it be possible to run the simulation a bit longer (e.g. 100 ps) so that we can actually use the equilibrium values of the `T`s and `u`s, which can be computed analytically?

Reply:
I chose this time because this is where one sees the largest discrepency between the new corrected implementation and the original implementation. If you ran until equilibrium, then the development branch will not fail the check. It is the relaxation to the equilibrium that is incorrect. Not the steady state solution.

Reply:
And yes. The development branch fails this test.